### PR TITLE
Add API for shareable drafts

### PR DESCRIPTION
**Pull request by jfreda**
This PR adds an API for sharing drafts documents with the Google Workspace organization.

### Breaking changes

- New required `google_workspace.domain` config attribute:
```hcl
google_workspace {
  // domain is the Google Workspace domain (e.g., "hashicorp.com").
  domain = "your-domain-dot-com"
}
```

### New API endpoints

- `/api/v1/drafts/{document_id}/shareable`
  - Methods: GET, PUT

### CODE REVIEW

File: internal/api/documents.go [hunk 1]
```diff
@@ -116,11 +125,19 @@ func DocumentHandler(
 			return
 		}
 
-		// Pass request off to the documents related resources handler if
-		// appropriate.
-		if isRelatedResourcesRequest {
+		// Pass request off to associated subcollection handler, if appropriate.
```
Comment (anubhavmishra) [inline internal/api/documents.go:128]: Could we define what a sub collection is in a comment? would be helpful context.

File: internal/cmd/commands/server/server.go [hunk 2]
```diff
@@ -157,6 +157,14 @@ func (c *Command) Run(args []string) int {
 			return 1
 		}
 	}
+	if cfg.GoogleWorkspace == nil {
+		c.UI.Error("error validating config: google_workspace block is required")
+		return 1
+	}
+	if cfg.GoogleWorkspace.Domain == "" {
+		c.UI.Error("error validating config: google_workspace domain is required")
```
Comment (anubhavmishra) [inline internal/cmd/commands/server/server.go:165]:
```suggestion
		c.UI.Error("error validating config: Google Workspace Domain is required")
```

We might want to be consistent with how we show these errors for invalid server config. For reference: https://github.com/hashicorp-forge/hermes/pull/261/files#diff-39d8b05de61be8fba5e909420b32a4182700e95200188a0be03efccc858000f0R200-R202

**Review by anubhavmishra — APPROVED** [review]
Minor comments and suggestions! Overall looks good!

File: internal/cmd/commands/server/server.go (hunk 2 above)
Comment (jfreda) [reply internal/cmd/commands/server/server.go:165]: Good call, I'll actually move this under that part you referenced. All of this needs to be refactored at some point, but that'll be more consistent for now.

Comment (anubhavmishra) [reply internal/cmd/commands/server/server.go:165]: Sounds good to me!

File: internal/api/documents.go (hunk 1 above)
Comment (jfreda) [reply internal/api/documents.go]: Done!

File: internal/cmd/commands/server/server.go (hunk 2 above)
Comment (jfreda) [reply internal/cmd/commands/server/server.go]: Done!

**Pull request merged**